app/src/main/java/com/hamdan/forzenbook/data/repository/LoginRepositoryImpl.kt

```diff
@@ -4,15 +4,18 @@ import com.hamdan.forzenbook.data.database.LoginDao
 import com.hamdan.forzenbook.data.network.LoginService
 
 class LoginRepositoryImpl(
+    // TODO implement the dao FA-84
     private val dao: LoginDao,
     private val service: LoginService
 ) : LoginRepository {
     override suspend fun requestValidation(email: String): Boolean {
         return service.requestValidation(email).isSuccessful
     }
 
-    // TODO Implement Getting the Token
     override suspend fun getToken(email: String, code: String): LoginData? {
-        TODO("Not yet implemented")
+        val response = service.getToken(email, code)
+        return if (response.isSuccessful) {
```

Comment:
Just throw if you don't have the correct data. No reason to return null

Reply:
So why even have this be nullable return value now?

Reply:
its a loginresponse now, but I thought the body could be null, it's now handled that if the body is null it will also throw an exception